MapleServer2/PacketHandlers/Game/InstrumentHandler.cs

```diff
@@ -150,12 +184,84 @@ private static void HandleCompose(GameSession session, PacketReader packet)
             session.Send(InstrumentPacket.Compose(item));
         }
 
+        private static void HandleStartEnsemble(GameSession session, PacketReader packet)
+        {
+            long instrumentItemUid = packet.ReadLong();
+            long scoreItemUid = packet.ReadLong();
+
+            Party party = GameServer.PartyManager.GetPartyById(session.Player.PartyId);
+            if (party == null)
+            {
+                return;
+            }
+
+            if (!session.Player.Inventory.Items.ContainsKey(scoreItemUid) || !session.Player.Inventory.Items.ContainsKey(instrumentItemUid))
+            {
+                return;
+            }
+
+
+            Item score = session.Player.Inventory.Items[scoreItemUid];
+
+            if (score.PlayCount <= 0)
+            {
+                return;
+            }
+
+            Item instrumentItem = session.Player.Inventory.Items[instrumentItemUid];
+            InsturmentInfoMetadata instrumentInfo = InstrumentInfoMetadataStorage.GetMetadata(instrumentItem.Function.Id);
+            InstrumentCategoryInfoMetadata instrumentCategory = InstrumentCategoryInfoMetadataStorage.GetMetadata(instrumentInfo.Category);
+            Instrument instrument = new Instrument(instrumentCategory.GMId, instrumentCategory.PercussionId, score.IsCustomScore, session.FieldPlayer.ObjectId)
+            {
+                Score = score,
+                Ensemble = true,
+                Improvise = false
+            };
+
+            session.Player.Instrument = session.FieldManager.RequestFieldObject(instrument);
+            session.Player.Instrument.Coord = session.FieldPlayer.Coord;
+
+            if (session.Player != party.Leader)
+            {
+                return;
+            }
+
+            int instrumentTick = session.ServerTick;
```

Comment:
If the servertick changes when the loop is happening that wouldn't cause some desync?

Reply:
Yep! it would change as the loop is happening, which is why we're storing the current tick in `instrumentTick`

Reply:
im dumb